packages/expo-module-template/android/src/main/java/{%= project.package %}/{%- project.name %}Module.kt

```diff
@@ -4,18 +4,44 @@ import expo.modules.kotlin.modules.Module
 import expo.modules.kotlin.modules.ModuleDefinition
 
 class <%- project.name %>Module : Module() {
+  // Each module class must implement the definition function. The definition consists of components
+  // that describe the module's functionality and behavior.
+  // See https://docs.expo.dev/modules/module-api for more details about available components.
   override fun definition() = ModuleDefinition {
+    // Sets the name of the module that JavaScript code will use to refer to the module. Takes a string as an argument.
+    // Can be inferred from module's class name, but it's recommended to set it explicitly for clarity.
+    // The module will be accessible from `requireNativeModule('<%- project.name %>')` in JavaScript.
     Name("<%- project.name %>")
 
-    AsyncFunction("helloAsync") { options: Map<String, String> ->
+    // Sets constant properties on the module. Can take a dictionary or a closure that returns a dictionary.
+    Constants(
+      "PI" to Math.PI
+    )
+
+    // Defines event names that the module can send to JavaScript.
+    Events("onChange")
+
+    // Defines a JavaScript function that always returns a Promise and whose native code
+    // is by default dispatched on the different thread than the JavaScript runtime runs on.
+    AsyncFunction("setValueAsync") { value: String ->
       println("Hello 👋")
+
+      // Send an event to JavaScript.
+      sendEvent("onChange", mapOf(
+        "value" to value
+      ))
     }
 
+    // Enables the module to be used as a view manager. The view manager definition is built from
+    // the definition components used in the closure passed to viewManager.
+    // Definition components that are accepted as part of the view manager definition: `View`, `Prop`.
     ViewManager {
+      // Defines the factory creating a native view, when the module is used as a view.
```

Comment:
```suggestion
      // Defines the factory creating a native view when the module is used as a view.
```
I'm not sure about that, but I think that comma is unnecessary here.